pkg/remediator/watch/filteredwatcher.go

```diff
@@ -109,17 +109,16 @@ var _ Runnable = &filteredWatcher{}
 // NewFiltered returns a new filtered watcher initialized with the given options.
 func NewFiltered(cfg watcherConfig) Runnable {
 	return &filteredWatcher{
-		gvk:                     cfg.gvk.String(),
-		startWatch:              cfg.startWatch,
-		resources:               cfg.resources,
-		queue:                   cfg.queue,
-		scope:                   cfg.scope,
-		syncName:                cfg.syncName,
-		base:                    watch.NewEmptyWatch(),
-		errorTracker:            make(map[string]time.Time),
-		conflictErrMap:          make(map[queue.GVKNN]status.ManagementConflictError),
-		addConflictErrorFunc:    cfg.addConflictErrorFunc,
-		removeConflictErrorFunc: cfg.removeConflictErrorFunc,
+		gvk:             cfg.gvk.String(),
+		startWatch:      cfg.startWatch,
+		resources:       cfg.resources,
+		queue:           cfg.queue,
+		scope:           cfg.scope,
+		syncName:        cfg.syncName,
+		base:            watch.NewEmptyWatch(),
+		errorTracker:    make(map[string]time.Time),
+		conflictErrMap:  make(map[queue.GVKNN]status.ManagementConflictError),
```

Comment:
Is this map something the ConflictHandler could handle? Does it need to be locked the same way?

Reply:
Seems redundant to store as both a map and a list, unless we care about the performance at scale. If performance and order are important, we could used an ordered map, like https://pkg.go.dev/github.com/wk8/go-ordered-map

Reply:
Switched to use ordered map.